client/my-sites/controller.js

```diff
@@ -71,7 +71,8 @@ function createNavigation( context ) {
 	}
 
 	return (
-		<NavigationComponent path={ context.path }
+		<AsyncLoad require="my-sites/navigation"
```

Comment:
I don't think we should async load navigation in general. I'd be ok with async loading the site picker alone when you switch sites.